ui/src/attestation/status.ts

```diff
@@ -0,0 +1,124 @@
+import { Signer, ethers } from "ethers";
+import * as svelteStore from "svelte/store";
+import type { Identity } from "../identity";
+import * as session from "../session";
+import { parseIdentitySha1 } from "../urn";
+
+import type { UnsealedSession } from "../session";
+
+import type { Wallet } from "../wallet";
+import { claimsAddress, ClaimsContract } from "./contract";
+
+export enum AttestationStatus {
+  Fetching = "Fetching",
+  Incomplete = "Incomplete",
+  Expired = "Expired",
+  UnmatchingEth = "UnmatchingEth",
+  UnmatchingRoot = "UnmatchingRoot",
+  Valid = "Valid",
+}
+
+export const attestationStatus: svelteStore.Writable<AttestationStatus> = svelteStore.writable(
+  AttestationStatus.Fetching
+);
+
+let watching = false;
+
+export async function watchAttestationStatus(
+  walletStore: svelteStore.Readable<Wallet>
+): Promise<void> {
+  if (watching) {
+    return;
+  }
+  watching = true;
+
+  const wallet = svelteStore.get(walletStore);
+  const claimWatcher = new ClaimWatcher(
```

Comment:
we should unwatch this :thinking:

Reply:
The `ClaimWatcher` unwatches and rewatches automatically based on the wallet address changes. We would need to explicitly unwatch only if we wanted to stop watching entirely and stop the polling loop below. It may make sense in the future, but now we have an infinite loop singleton always running in the back.